Core/XMPPStream.h

```diff
@@ -1032,6 +1124,9 @@ extern const NSTimeInterval XMPPStreamTimeoutNone;
  * These methods are called after their respective XML elements are sent over the stream.
  * These methods may be used to listen for certain events (such as an unavailable presence having been sent),
  * or for general logging purposes. (E.g. a central history logging mechanism).
+ *
+ * Delegates can obtain event metadata associated with the respective element by calling @c currentElementEvent on @c sender
+ * from within these callbacks. For more details, please refer to @c XMPPElementEvent documentation.
```

Comment:
As an extension to my comment above about adding XMPPElementEvent to the callbacks. Could you create another delegate protocol to complement XMPPStreamDelegate and put it in another file? For instance:

```objc
@protocol XMPPElementEventDelegate <NSObject>
 - (void)xmppStream:(XMPPStream *)sender didSendIQ:(XMPPIQ *)iq elementEvent:(XMPPElementEvent*)elementEvent;
 - (void)xmppStream:(XMPPStream *)sender didSendMessage:(XMPPMessage *)message elementEvent:(XMPPElementEvent*)elementEvent;
// etc..
 - (void)xmppStream:(XMPPStream *)sender didFinishProcessingElementEvent:(XMPPElementEvent *)event NS_SWIFT_NAME(xmppStream(_:didFinishProcessing:));
@end